mautic/mautic
# [DPMMA-2044] Loading custom fonts into GrapesJS editor - plugin

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [No]
| New feature/enhancement? (use the a.x branch)      | [Yes]
| Deprecations?                          | [No]
| BC breaks? (use the c.x branch)        | [No]
| Automated tests included?              | [No] 
| Related user documentation PR URL      | #
| Related developer documentation PR URL | #
| Issue(s) addressed                     | 

Related PR in grapesjs-preset-mautic: [https://github.com/mautic/grapesjs-preset-mautic/pull/43](https://github.com/mautic/grapesjs-preset-mautic/pull/43
)
#### Description:

PR introduces the ability to load and use custom fonts defined in the Mautic configuration to the list of fonts in the Style Manager of the GrapesJS builder. Only fonts with a unique name are added to the selection. Style links for all fonts are added to the canvas, but when saving a message or landing page, only those actually used to style the document are finally added in the HTML <head>.

![fonts](https://user-images.githubusercontent.com/102536220/223717276-8fa7696f-fca8-47ed-b2cc-eb3eb79aa535.jpg)

#### Steps to test this PR:



1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Define custom editor_fonts in Mautic configuration (config.php/local.php/...), e.g.
```
(...)
'editor_fonts' => array(
        '0' => array(
            'name' => 'Smokum',
            'font' => 'Smokum, cursive',
            'url' => 'https://fonts.googleapis.com/css2?family=Smokum&display=swap'
        ),
        '1' => array(
            'name' => 'Sofia',
            'font' => 'Sofia, sans-serif',
            'url' => 'https://fonts.googleapis.com/css?family=Sofia'
        )
),
(...)
```
3. Open GrapesJS builder with email or landing page. 
4. Select a block of text and open Style Manager -> Typography -> Font. New fonts should be available to choose from.
5. Check that the font selection allows you to style the selected text and that the changes are correctly saved and displayed in the page preview or in the sent message.
